views/templates/hook/displayAdminOrderLeft.tpl

```diff
@@ -22,153 +22,31 @@
-                    <label for="orderPayPalRefundAmount{$orderPayPalTransaction.id|escape:'html':'UTF-8'}">{l s='Choose amount to refund (tax included)' mod='ps_checkout'}</label>
-                    <div class="input-group">
-                      <input name="orderPayPalRefundAmount" id="orderPayPalRefundAmount{$orderPayPalTransaction.id|escape:'html':'UTF-8'}" type="number" step=".01" min="0" max="{$orderPayPalTransaction.maxAmountRefundable|escape:'html':'UTF-8'|string_format:"%.2f"}" value="{$orderPayPalTransaction.maxAmountRefundable|escape:'html':'UTF-8'|string_format:"%.2f"}" class="form-control">
-                      <div id="orderPayPalRefundCurrencyContainer" class="input-group-addon">{$orderPayPalTransaction.currency|escape:'html':'UTF-8'}</div>
-                    </div>
-                  </div>
-                  <button type="submit" name="orderPayPalRefundSubmit" class="btn btn-primary">
-                    <i class="icon-exchange"></i>
-                    {l s='Refund' mod='ps_checkout'}
-                  </button>
-                </form>
-                <span class="help-block">
-                  {l s='Maximum [AMOUNT_MAX] [CURRENCY] (tax included)' sprintf=['[AMOUNT_MAX]' => $orderPayPalTransaction.maxAmountRefundable|escape:'html':'UTF-8'|string_format:"%.2f", '[CURRENCY]' => $orderPayPalTransaction.currency|escape:'html':'UTF-8'] mod='ps_checkout'}
-                </span>
-              </td>
-            </tr>
-          {/if}
-        {/foreach}
-        </tbody>
-      </table>
-    </div>
-  {/if}
 </div>
 
 <script>
-  $(document).ready(function() {
-    $(document).on('submit', '.orderPayPalRefundForm', function () {
-      let isApproved = confirm("{l s='Your transaction refund request will be sent to PayPal. After that, you\'ll still need to manually apply the refund registration in the PrestaShop order, to choose the type of refund and generate invoice.' mod='ps_checkout'}");
-
-      if (isApproved) {
-        $('button[name="orderPayPalRefundSubmit"]').html('<i class="process-icon-loading"></i>').prop('disabled', true);
-      }
-
-      return isApproved;
+  if (undefined !== ps_checkout) {
+    ps_checkout.initialize({
+      legacy: 1,
+      orderPrestaShopId: {$orderPrestaShopId|intval},
+      orderPayPalBaseUrl: '{$orderPayPalBaseUrl|escape:'javascript'}',
+      orderPayPalContainer: '.paypal-order-container',
+      orderPayPalLoaderContainer: '.paypal-order-loader',
+      orderPayPalNotificationsContainer: '.paypal-order-notifications',
+      orderPayPalRefundButton: '#ps_checkout button.refund',
+      orderPayPalModalContainerPrefix: '#ps-checkout-refund-',
+      orderPayPalModalContainer: '.ps-checkout-refund',
+      orderPayPalModalNotificationsContainer: '.modal-notifications',
+      orderPayPalModalContentContainer: '.modal-content-container',
+      orderPayPalModalLoaderContainer: '.modal-loader',
+      orderPayPalModalRefundForm: '.ps-checkout-refund-form',
     });
-  });
+  }
```

Comment:
To avoid duplicated code between this file and views/templates/hook/displayAdminOrderMainBottom.tpl, can you move this part in another file which can be included? https://www.smarty.net/docsv2/en/language.function.include.tpl

Reply:
How manage `legacy` argument not the same ?

Reply:
you may send it as another parameter